src/components/Aframe/my-loader.js

```diff
@@ -309,18 +307,24 @@ AFRAME.registerComponent("myloader", {
       currentVolume = undefined;
     }
 
-    if (this.data.volumeData !== "") {
+    if (volueDataObject.path !== "") {
       this.hiddenLabel.style.display = "";
       var el = this.el;
       var data = this.data;
-      //var transferTexture = this.transferTexture ;
-      var myWidth = this.myCanvas.width;
-      var myheight = this.myCanvas.height;
+      var canvasWidth = this.myCanvas.width;
+      var canvasHeight = this.myCanvas.height;
       var colorMap = null;
       var useTransferFunction;
       var hiddenLabel = this.hiddenLabel;
       var enabledColorMapping = this.colorMapEnabled;
-      var iam = this;
+
+      const updateColorMapping = this.updateColorMapping;
+      const updateTransferTexture = this.updateTransferTexture;
+
+      var x_dim = volueDataObject.x_spacing;
```

Comment:
if you're okay using the same variable names as in the options object, you could instead do
```js
const {x_spacing, y_spacing, z_spacing, slices} = volumeDataOptions;
```

Reply:
that could even be done further up in the method so you could pull out `path` too (if anything doesn't exist it will default to `undefined`

Reply:
Done !!